src/MICore/CommandFactories/MICommandFactory.cs

```diff
@@ -418,7 +418,7 @@ public virtual async Task Terminate()
 
         #region Breakpoints
 
-        protected virtual StringBuilder BuildBreakInsert(string condition, bool enabled)
+        protected virtual Task<StringBuilder> BuildBreakInsert(string condition, bool enabled)
```

Comment:
what if you call this from `GetBreakInsertMainCommand` so that there isn't duplicate `-break-insert` and checking for the `on` keyword?